Current source does not compile

```$go get -u github.com/dadleyy/marlow/marlowc
# github.com/dadleyy/marlow/marlowc
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:56:3: undefined: mpb.Output
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:64:30: cannot use len(name) (type int) as type decor.WC in argument to decor.StaticName
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:64:38: cannot use 0 (type int) as type decor.WC in argument to decor.StaticName
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:65:4: undefined: decor.ETA
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:67:41: cannot use 5 (type int) as type decor.WC in argument to decor.Percentage
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:67:44: cannot use 0 (type int) as type decor.WC in argument to decor.Percentage
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:111:6: bar.Incr undefined (type *mpb.Bar has no field or method Incr)
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:114:5: bar.Complete undefined (type *mpb.Bar has no field or method Complete)
vendor/src/github.com/dadleyy/marlow/marlowc/main.go:115:10: progress.Stop undefined (type *mpb.Progress has no field or method Stop)```

Possibly related to being dependent upon latest version of dependencies.

--- a/examples/library/models/author_test.go
+++ b/examples/library/models/author_test.go
@@ -227,7 +227,7 @@ func Test_Author(t *testing.T) {
 
 			g.Assert(authors[0].Name).Equal("learned author")
 			g.Assert(authors[0].UniversityID.Valid).Equal(true)
-			g.Assert(authors[0].UniversityID.Int64).Equal(10)
+			g.Assert(authors[0].UniversityID.Int64).Equal(int64(10))
 
 			g.Assert(authors[1].Name).Equal("other author")
 			g.Assert(authors[1].UniversityID.Valid).Equal(false)
@@ -264,7 +264,7 @@ func Test_Author(t *testing.T) {
 			})
 			g.Assert(e).Equal(nil)
 			g.Assert(len(results)).Equal(1)
-			g.Assert(results[0].Int64).Equal(10)
+			g.Assert(results[0].Int64).Equal(int64(10))
 			store.DeleteAuthors(&AuthorBlueprint{
 				NameLike: []string{"%limited%"},
 			})
@@ -286,7 +286,7 @@ func Test_Author(t *testing.T) {
 			g.Assert(e).Equal(nil)
 			g.Assert(len(results)).Equal(1)
 			g.Assert(results[0].Valid).Equal(true)
-			g.Assert(results[0].Int64).Equal(10)
+			g.Assert(results[0].Int64).Equal(int64(10))
 		})
 
 		g.It("allows consumers to select individual author names with limit", func() {
@@ -347,19 +347,19 @@ func Test_Author(t *testing.T) {
 		g.It("allows the consumer to update the author id", func() {
 			c, e := store.UpdateAuthorID(1991, &AuthorBlueprint{ID: []int{8}})
 			g.Assert(e).Equal(nil)
-			g.Assert(c).Equal(1)
+			g.Assert(c).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the author university id using nil", func() {
 			c, e := store.UpdateAuthorUniversityID(nil, &AuthorBlueprint{ID: []int{1}})
 			g.Assert(e).Equal(nil)
-			g.Assert(c).Equal(1)
+			g.Assert(c).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the author university id using valid: false", func() {
 			c, e := store.UpdateAuthorUniversityID(&sql.NullInt64{Valid: false}, &AuthorBlueprint{ID: []int{2}})
 			g.Assert(e).Equal(nil)
-			g.Assert(c).Equal(1)
+			g.Assert(c).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the author university id using valid: true", func() {
@@ -369,7 +369,7 @@ func Test_Author(t *testing.T) {
 			}, &AuthorBlueprint{ID: []int{2}})
 
 			g.Assert(e).Equal(nil)
-			g.Assert(c).Equal(1)
+			g.Assert(c).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the author name", func() {
@@ -379,7 +379,7 @@ func Test_Author(t *testing.T) {
 
 			updatedCount, e := store.UpdateAuthorName("danny", &AuthorBlueprint{ID: []int{1}})
 			g.Assert(e).Equal(nil)
-			g.Assert(updatedCount).Equal(1)
+			g.Assert(updatedCount).Equal(int64(1))
 
 			a, e := store.FindAuthors(&AuthorBlueprint{ID: []int{1}})
 			g.Assert(e).Equal(nil)
@@ -398,7 +398,7 @@ func Test_Author(t *testing.T) {
 			g.It("returns immediately with 0 if no authors", func() {
 				s, e := store.CreateAuthors()
 				g.Assert(e).Equal(nil)
-				g.Assert(s).Equal(0)
+				g.Assert(s).Equal(int64(0))
 			})
 
 			g.It("returns the id of the latest author created", func() {
@@ -600,19 +600,19 @@ func Test_Author(t *testing.T) {
 			g.It("returns an error and a negative number with an empty blueprint", func() {
 				c, e := store.DeleteAuthors(&AuthorBlueprint{})
 				g.Assert(e == nil).Equal(false)
-				g.Assert(c).Equal(-1)
+				g.Assert(c).Equal(int64(-1))
 			})
 
 			g.It("returns an error and a negative number without a blueprint", func() {
 				c, e := store.DeleteAuthors(nil)
 				g.Assert(e == nil).Equal(false)
-				g.Assert(c).Equal(-1)
+				g.Assert(c).Equal(int64(-1))
 			})
 
 			g.It("successfully deletes the records found by the blueprint", func() {
 				deleted, e := store.DeleteAuthors(&AuthorBlueprint{ID: []int{13}})
 				g.Assert(e).Equal(nil)
-				g.Assert(deleted).Equal(1)
+				g.Assert(deleted).Equal(int64(1))
 				found, e := store.CountAuthors(&AuthorBlueprint{ID: []int{13}})
 				g.Assert(e).Equal(nil)
 				g.Assert(found).Equal(0)

--- a/examples/library/models/book_test.go
+++ b/examples/library/models/book_test.go
@@ -260,7 +260,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{1},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the author id", func() {
@@ -268,7 +268,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{1},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the book year published", func() {
@@ -276,7 +276,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{10},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the book series id (valid: false)", func() {
@@ -286,7 +286,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{10},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the book series id (non-nil)", func() {
@@ -297,7 +297,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{10},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the book series id (nil)", func() {
@@ -305,7 +305,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{10},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to update the book id", func() {
@@ -313,7 +313,7 @@ func Test_Book(t *testing.T) {
 				ID: []int{10},
 			})
 			g.Assert(e).Equal(nil)
-			g.Assert(results).Equal(1)
+			g.Assert(results).Equal(int64(1))
 		})
 
 		g.It("allows the consumer to select explicit year published", func() {
@@ -330,25 +330,25 @@ func Test_Book(t *testing.T) {
 			g.It("returns an error and a negative number with an empty blueprint", func() {
 				c, e := store.DeleteBooks(&BookBlueprint{})
 				g.Assert(e == nil).Equal(false)
-				g.Assert(c).Equal(-1)
+				g.Assert(c).Equal(int64(-1))
 			})
 
 			g.It("returns an error and a negative number without a blueprint", func() {
 				c, e := store.DeleteBooks(nil)
 				g.Assert(e == nil).Equal(false)
-				g.Assert(c).Equal(-1)
+				g.Assert(c).Equal(int64(-1))
 			})
 
 			g.It("successfully returns 0 if no books were found to delete", func() {
 				deleted, e := store.DeleteBooks(&BookBlueprint{ID: []int{-1000}})
 				g.Assert(e).Equal(nil)
-				g.Assert(deleted).Equal(0)
+				g.Assert(deleted).Equal(int64(0))
 			})
 
 			g.It("successfully deletes the records found by the blueprint", func() {
 				deleted, e := store.DeleteBooks(&BookBlueprint{ID: []int{13}})
 				g.Assert(e).Equal(nil)
-				g.Assert(deleted).Equal(1)
+				g.Assert(deleted).Equal(int64(1))
 				found, e := store.CountBooks(&BookBlueprint{ID: []int{13}})
 				g.Assert(e).Equal(nil)
 				g.Assert(found).Equal(0)
@@ -359,7 +359,7 @@ func Test_Book(t *testing.T) {
 			g.It("returns immediately with 0 if no authors", func() {
 				s, e := store.CreateBooks()
 				g.Assert(e).Equal(nil)
-				g.Assert(s).Equal(0)
+				g.Assert(s).Equal(int64(0))
 			})
 
 			g.It("returns the number of authors created", func() {

--- a/examples/library/models/genre_test.go
+++ b/examples/library/models/genre_test.go
@@ -141,7 +141,7 @@ func Test_Genre(t *testing.T) {
 					g.Assert(e).Equal(nil)
 					g.Assert(len(parents)).Equal(1)
 					g.Assert(parents[0].Valid).Equal(true)
-					g.Assert(parents[0].Int64).Equal(10)
+					g.Assert(parents[0].Int64).Equal(int64(10))
 				})
 
 				g.It("allows counting w/ empty NullInt64 blueprint (NOT NULL)", func() {
@@ -241,7 +241,7 @@ func Test_Genre(t *testing.T) {
 					g.Assert(e).Equal(nil)
 					g.Assert(len(ids)).Equal(1)
 					g.Assert(ids[0].Valid).Equal(true)
-					g.Assert(ids[0].Int64).Equal(100)
+					g.Assert(ids[0].Int64).Equal(int64(100))
 
 					p.Scan(nil)
 					_, e = store.UpdateGenreParentID(&p, bp)

--- a/glide.lock
+++ b/glide.lock
@@ -1,21 +1,28 @@
-hash: f01968c098f896b4e9b97fa4b38e8ee3e13ad8c8aa1daab736d872f22463a571
-updated: 2018-02-25T11:55:29.759498125-05:00
+hash: 87dde26b983ed5655c1758c11a23bf90293367fd1609ebe52f7f52f2aef62bd1
+updated: 2018-06-26T09:55:56.053333896-04:00
 imports:
 - name: github.com/dustin/go-humanize
   version: bb3d318650d48840a39aa21a027c6630e198e626
 - name: github.com/franela/goblin
-  version: 74c9fe110d4bfd04c222a089a309e0a97e258534
+  version: cd5d08fb4ede9eaac1812fdb513552e7404eae2e
 - name: github.com/gedex/inflector
   version: 046f2c31204676e3623730ee70fd0964e05822e7
+- name: github.com/lib/pq
+  version: b609790bd85edf8e9ab7e0f8912750a786177bcf
+  subpackages:
+  - oid
 - name: github.com/mattn/go-isatty
   version: 56b76bdf51f7708750eac80fa38b952bb9f32639
 - name: github.com/mattn/go-sqlite3
   version: ca5e3819723d8eeaf170ad510e7da1d6d2e94a08
 - name: github.com/vbauerster/mpb
-  version: dac08dfacedfef1da943f0491a430e4a8edf5bb0
+  version: 3674fd1f0789dbb003aa91a7b06138df0ef83c0f
   subpackages:
   - cwriter
   - decor
+  - internal
+- name: github.com/VividCortex/ewma
+  version: 43880d236f695d39c62cf7aa4ebd4508c258e6c0
 - name: golang.org/x/net
   version: 8e573f4005aa312856df2ea97c32b9beac70dd89
   subpackages:
@@ -24,8 +31,4 @@ imports:
   version: 62bee037599929a6e9146f29d10dd5208c43507d
   subpackages:
   - unix
-testImports:
-- name: github.com/lib/pq
-  version: b609790bd85edf8e9ab7e0f8912750a786177bcf
-  subpackages:
-  - oid
+testImports: []

--- a/marlowc/main.go
+++ b/marlowc/main.go
@@ -4,14 +4,15 @@ import "os"
 import "io"
 import "fmt"
 import "flag"
-import "time"
 import "path"
+import "sync"
 import "bytes"
 import "strings"
 import "go/build"
+
 import "github.com/vbauerster/mpb"
 import "github.com/dustin/go-humanize"
-import "github.com/vbauerster/mpb/decor"
+
 import "github.com/dadleyy/marlow/marlow"
 import "github.com/dadleyy/marlow/marlow/constants"
 
@@ -44,7 +45,9 @@ func main() {
 		progressOut = os.Stdout
 	}
 
-	total, name, done := len(sourceFiles), fmt.Sprintf("compiling files"), make(chan struct{})
+	fmt.Fprintf(progressOut, "starting progress")
+
+	total := len(sourceFiles)
 
 	// If no files were found, exit.
 	if total == 0 {
@@ -53,19 +56,10 @@ func main() {
 
 	// Create the progress bar.
 	progress := mpb.New(
-		mpb.Output(progressOut),
-		mpb.WithWidth(100),
-		mpb.WithRefreshRate(time.Millisecond),
-		mpb.WithShutdownNotifier(done),
+		mpb.WithOutput(progressOut),
+		mpb.WithWaitGroup(&sync.WaitGroup{}),
 	)
-
-	bar := progress.AddBar(int64(total),
-		mpb.PrependDecorators(
-			decor.StaticName(name, len(name), 0),
-			decor.ETA(4, decor.DSyncSpace),
-		),
-		mpb.AppendDecorators(decor.Percentage(5, 0)),
-	)
+	bar := progress.AddBar(int64(total))
 
 	// Keep a list of files that have been created to print out at the end.
 	results := make(map[string]int64)
@@ -73,6 +67,7 @@ func main() {
 	for _, name := range sourceFiles {
 		// Skip files that have already bee compiled.
 		if strings.HasSuffix(path.Base(name), options.ext) {
+			bar.IncrBy(1)
 			continue
 		}
 
@@ -99,8 +94,11 @@ func main() {
 		// If no data was copied we had an no-op gen source, remove the file and continue.
 		if size == 0 {
 			os.Remove(options.generatedName(name))
+			bar.IncrBy(1)
 			continue
 		}
+
+		fmt.Fprintf(os.Stdout, "completed compilation of %s\n", name)
 
 		results[options.generatedName(name)] = size
 
@@ -108,12 +106,10 @@ func main() {
 		writer.Close()
 
 		// Let our progress bar know we're done.
-		bar.Incr(1)
+		bar.IncrBy(1)
 	}
 
-	bar.Complete()
-	progress.Stop()
-	<-done
+	progress.Wait()
 
 	// If no files were the target of compilation, or the silent flag was used, do nothing.
 	if len(results) == 0 || options.silent == true {

